src/test/java/qna/domain/QuestionTest.java

```diff
@@ -0,0 +1,70 @@
+package qna.domain;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.DisplayNameGeneration;
+import org.junit.jupiter.api.DisplayNameGenerator;
+import org.junit.jupiter.api.Test;
+import qna.exception.CannotDeleteException;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static qna.fixture.Fixture.*;
+
+@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+@SuppressWarnings("NonAsciiCharacters")
+class QuestionTest {
+
+    @Test
+    void 질문자는_질문을_삭제할_수_있다() {
+        // given
+        final User gugu = new User("gugu", "password", "구구", "gugu@gugu.com");
+        final Question question = new Question("제목", "내용").writeBy(gugu);
+
+        // when
+        question.deleteBy(gugu);
+
+        // then
+        assertThat(question.isDeleted()).isTrue();
+    }
+
+    @Test
+    void 질문_작성자가_아닌_다른이가_삭제할_경우_예외가_발생한다() {
+        // given
+        final User gugu = new User("gugu", "password", "구구", "gugu@gugu.com");
+        final Question question = new Question("제목", "내용").writeBy(gugu);
+
+        // expect
+        assertThatThrownBy(() -> question.deleteBy(JAVAJIGI))
+                .isInstanceOf(CannotDeleteException.class)
+                .hasMessageContaining("질문을 삭제할 권한이 없습니다.");
+    }
+
+    @Test
+    void 질문자와_답변자들이_동일하면_삭제가_가능하다() {
+        // given
+        final User gugu = new User("gugu", "password", "구구", "gugu@gugu.com");
+        final Question question = new Question("제목", "내용").writeBy(gugu);
+        question.addAnswer(new Answer(gugu, question, "레벨 4 강의는 제가 합니다"));
```

Comment:
```suggestion
        question.addAnswer(gugu, question, "레벨 4 강의는 제가 합니다");
```

테스트 코드를 작성하는 입장에서 보니까 이렇게 쓸 수 있다면 더 편하겠네요. 어떻게 생각하시나요?

Reply:
명시적으로 Answer가 들어가야함을 보여줄 필요성도 있는 것 같아 이전 방법도 나쁘지 않다고 생각했습니다

일단 반영했습니다! 장단점은 차차 느껴보도록 하겠습니다